k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java

```diff
@@ -470,6 +473,13 @@ public void setCharset(String charset) throws MessagingException {
         }
     }
 
+    public boolean isHighPriority() {
+        if (getFirstHeader(highPriorityTag)!=null && getFirstHeader(highPriorityTag).equals(xPriorityContent)) {
```

Comment:
you can check if `xPriorityContent.equals(tag)`, which will also return false if the header is null